lib/matplotlib/axis.py

```diff
@@ -300,33 +300,30 @@ def _apply_params(self, **kw):
         switches = [k for k in kw if k in switchkw]
         for k in switches:
             setattr(self, k, kw.pop(k))
-        dirpad = [k for k in kw if k in ['pad', 'tickdir']]
-        if dirpad:
+        newmarker = [k for k in kw if k in ['size', 'width', 'pad', 'tickdir']]
+        if newmarker:
+            self._size = kw.pop('size', self._size)
+            self._width = kw.pop('width', self._width)
             self._base_pad = kw.pop('pad', self._base_pad)
+            # apply_tickdir makes _pad and _tickmarkers
             self.apply_tickdir(kw.pop('tickdir', self._tickdir))
+            self.tick1line.set_marker(self._tickmarkers[0])
+            self.tick2line.set_marker(self._tickmarkers[1])
+            for line in (self.tick1line, self.tick2line):
+                line.set_markersize(self._size)
+                line.set_markeredgewidth(self._width)
+            # _get_text1_transform uses _pad from apply_tickdir
```

Comment:
I think you should add a comment mentioning that `apply_tickdir` uses `self._size`  to compute the `self._pad` value, as this was the core of the problem here.